### PR TITLE
ON-31943 # `useSignUp` hook

### PR DESCRIPTION
**Pull request by RyanButton**

### CODE REVIEW

**Comment by kizaonline** [comment]
`useFormsAppUserSignUp.ts`?

**Comment by RyanButton** [comment]
I don't mind that name, I just wanted it to mirror `useLogin.ts` is all

File: src/hooks/useSignUp.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      signUpError: null,
+    }))
+
+    try {
+      const newUser = await authService.signUp({
+        formsAppId,
+        email: username,
+        firstName,
+        lastName,
+        generatePassword: true,
```
Comment (kizaonline) [inline src/hooks/useSignUp.ts:66]: Whys this always true?

Comment (RyanButton) [reply src/hooks/useSignUp.ts]: a doozie

**Review by kizaonline — APPROVED** [review]

**Pull request merged**